src/chaincode/dex/createPool.ts

```
@@ -45,8 +45,26 @@ export async function createPool(ctx: GalaChainContext, dto: CreatePoolDto): Pro
     protocolFee = protocolFeeConfig.protocolFee;
   }
 
-  // Create pool
-  const pool = new Pool(token0, token1, dto.token0, dto.token1, dto.fee, dto.initialSqrtPrice, protocolFee);
+  // Create pool with private pool settings
+  const isPrivate = dto.isPrivate ?? false;
+  const whitelist = dto.whitelist ?? [];
+  const creator = ctx.callingUser;
+
+  // Ensure creator is in whitelist for private pools
+  const finalWhitelist = isPrivate && !whitelist.includes(creator) ? [creator, ...whitelist] : whitelist;
```

(optional) this is an implicit behavior - adding the creator. Maybe instead it should throw an error if the creator is missing?

---

I think I'm ok with defaulting to the calling user for this one.